### PR TITLE
LUCENE-10243: increase unicode versions of tokenizers to 12.1

### PR DESCRIPTION
**Pull request by rmuir**
Increase the unicode versions of our tokenizers from `9` to `12.1`.
Modify jflex grammars appropriately for changes to UAX#29 between these versions.
Modify/regenerate conformance tests for changes between these versions.

### CODE REVIEW

**Comment by rmuir** [comment]
I'm doing this with a few commits. For me, it makes sense to first bump the versions and the tests, then iterate on any grammar refactoring as separate changes (keeping the tests passing, being able to see diffs in generated code, etc).

File: gradle/generation/icu.gradle [hunk 1]
```diff
@@ -283,35 +283,3 @@ configure(project(":lucene:analysis:common")) {
 
   regenerate.dependsOn wrapWithPersistentChecksums(generateUnicodePropsInternal, [ andThenTasks: ["spotlessJava", "spotlessJavaApply"] ])
 }
-
```
Comment (dweiss) [inline gradle/generation/icu.gradle:286]: Awesome that it's gone. :)

File: lucene/analysis/common/src/java/org/apache/lucene/analysis/email/UAX29URLEmailTokenizerImpl.jflex [hunk 2]
```diff
@@ -61,8 +61,10 @@ ExtFmtZwj           = [\p{WB:Format}\p{WB:Extend}\p{WB:ZWJ}]*
 //////////////////////////////////////////////////////////////////////////
 // Begin Emoji Macros - see documentation below, near the EMOJI_TYPE rule
 
-// TODO: Remove this include file when JFlex supports these properties directly (in Unicode 11.0+)
```
Comment (dweiss) [inline lucene/analysis/common/src/java/org/apache/lucene/analysis/email/UAX29URLEmailTokenizerImpl.jflex:64]: Again, great so see this include gone.

**Review by dweiss — APPROVED** [review]

**Comment by rmuir** [comment]
So I think this is pretty close. We still "tweak" the UAX#29 by incorporating UTS#51 sequence grammars to deal with emoji. This difference (see "Instead of" in the grammar files for more explanation), is why two conformance tests are disabled.

We could probably do a few more cosmetic tweaks to the grammars, but I think for the most part it is cleaner already and we can iterate.

**Pull request merged**